src/audio.c

```diff
@@ -185,8 +194,10 @@ static void audio_callback()
  *            The frequency in Hz to play back samples at
  * @param[in] numbuffers
  *            The number of buffers to allocate internally
+ * @param[in] fill_buffer_callback
+ *            A function to be called when more sample data is needed
  */
-void audio_init(const int frequency, int numbuffers)
+void audio_init(const int frequency, int numbuffers, audio_fill_buffer_callback fill_buffer_callback)
```

Comment:
There's the dfsdemo. So I guess I should update its call to `audio_init`. Or possibly create a new function called `audio_set_buffer_callback` or something like that. This is just the way I implemented it back when I originally made these changes, which was a while ago (when I had begun working on this: https://www.youtube.com/watch?v=EqrgPTFNXAA).